s3_tagger.py

```diff
@@ -0,0 +1,180 @@
+def get_s3():
+    s3_client = boto3.client("s3")
+    return s3_client
+
+
+def get_objects_in_prefix(s3_bucket, s3_prefix, s3_client):
+    objects_in_prefix = s3_client.list_objects(Bucket=s3_bucket, Prefix=s3_prefix)[
+        "Contents"
+    ]
+    objects_to_tag = []
+    for key in objects_in_prefix:
+        if "$folder$" not in key["Key"]:
+            objects_to_tag.append(key["Key"])
+    return objects_to_tag
+
+
+def tag_path(objects_to_tag, s3_client, s3_bucket, csv_data):
+
+    logger.info(f"Found {len(objects_to_tag)} objects to tag in specified path")
+    tagged_objects = 0
+    for row in objects_to_tag:
+        is_tagged = tag_object(row, s3_client, s3_bucket, csv_data)
+        tagged_objects = tagged_objects + is_tagged
+
+    if tagged_objects == 0:
+        logger.warning("No objects tagged")
+    else:
+        logger.info(f"{tagged_objects} objects were tagged")
+
+
+def get_parameters():
+    parser = argparse.ArgumentParser(
+        description="A Python script which receives three args. 1. CSV location. 2. S3 Bucket. 3. S3-prefix, "
+        "It will then tag all the objects in the prefix as described in the CSV supplied"
+    )
+
+    # Parse command line inputs and set defaults
+    parser.add_argument("--csv_location", help="The location of the CSV file to parse")
+    parser.add_argument("--bucket", help="The bucket to tag")
+    parser.add_argument(
+        "--s3_prefix", help="The path to crawl through where objects need to be tagged"
+    )
+    parser.add_argument("--log-level", default="INFO")
+
+    _args = parser.parse_args()
+
+    # Override arguments with environment variables where set
+    if "csv_location" in os.environ:
+        _args.csv_location = os.environ["csv_location"]
+
+    if "bucket" in os.environ:
+        _args.bucket = os.environ["bucket"]
+
+    if "s3_prefix" in os.environ:
+        _args.s3_prefix = os.environ["s3_prefix"]
+
+    if "log_level" in os.environ:
+        _args.log_level = os.environ["log_level"]
+
+    required_args = ["csv_location", "bucket", "s3_prefix"]
+    missing_args = []
+
+    for required_message_key in required_args:
+        if required_message_key not in _args:
+            missing_args.append(required_message_key)
+
+    if missing_args:
+        raise argparse.ArgumentError(
+            None,
+            "ArgumentError: The following required arguments are missing: {}".format(
+                ", ".join(missing_args)
+            ),
+        )
+
+    return _args
+
+
+if __name__ == "__main__":
+    args = get_parameters()
+
+    # remove tailing or leading / from prefix
+    s3_prefix = args.s3_prefix
+    if s3_prefix.startswith("/"):
+        s3_prefix = s3_prefix.lstrip("/")
+    if s3_prefix.endswith("/"):
+        s3_prefix = s3_prefix.rstrip("/")
+
+    s3 = get_s3()
+    csv_data = read_csv(args.csv_location, s3)
+    objects_to_tag = get_objects_in_prefix(args.bucket, s3_prefix, s3)
+    tag_path(objects_to_tag, s3, args.bucket, csv_data)
```

Comment:
For these lines, can you add high level log messages please?
ie. "Retrieving S3 client", "Reading CSV file in location X"

It will help trace any issues once in production.